shub/deploy.py

```diff
@@ -140,17 +141,17 @@ def _upload_egg(target, eggpath, project, version, auth):
     return make_deploy_request(url, data, files, auth)
 
 
-def _build_egg():
+def _build_egg(setup_file):
     closest = closest_scrapy_cfg()
     os.chdir(os.path.dirname(closest))
-    if not os.path.exists('setup.py'):
+    if not os.path.exists(setup_file):
```

Comment:
Well, I think that adding options doesn't have to add complexity to the tool usage if we provide nice and logic defaults for that options. On the other hand, I agree with @eliasdorneles that this particular feature can be unnecessary if packages to be deploy are well planned and follow standards. BUT, if this feature doesn't affect the way users use the tool, and it doesn't break backwards compatibility, it provides a bit more of flexibility to the tool.
@josericardo That is right. Warning messages are usually ignored. So, the proposed behaviour is to throw an error if a setup file is given and it doesn't exist. Right?